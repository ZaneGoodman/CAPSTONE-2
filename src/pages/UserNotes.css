.UserNotes-container {
  background-image: url("https://images.freeimages.com/images/large-previews/b14/rosary-9-1244995.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  position: absolute;
  margin: 0;
  padding: 0;
  left: 0;
  top: 49px;
  bottom: 0;
  right: 0;
  overflow: scroll;
}
.UserNotes-list {
  text-align: center;
  margin: auto;
  margin-top: 20px;
}
.UserNotes-note-container {
  margin: auto;
  width: 30rem;
}
