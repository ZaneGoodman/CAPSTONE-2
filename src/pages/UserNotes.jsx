import React, { useEffect, useState } from "react";
import Notes from "../models/notes";
import NoteInfo from "./NoteInfo";
import { useAuth } from "../provider/authProvider";
import { v4 as uuid } from "uuid";

const UserNotes = () => {
  const { username } = useAuth();

  const [notes, setNotes] = useState([]);
  //Get all notes that the user has submitted to their profile and render them
  useEffect(() => {
    async function getUserNotes() {
      const allNotes = await Notes.getAllNotes(username);
      setNotes(() => allNotes.data.allNotes);
    }
    getUserNotes();
  }, [username]);

  return (
    <>
      <h1>Your Prayer Notes</h1>
      <ul>
        {notes.map((note) => (
          <li key={uuid()}>
            <NoteInfo
              date={note.date}
              season={note.season}
              has_prayed={note.has_prayed}
              notes={note.notes}
            />
          </li>
        ))}
      </ul>
    </>
  );
};

export default UserNotes;
