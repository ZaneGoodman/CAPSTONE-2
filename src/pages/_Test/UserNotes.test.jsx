import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import AuthProvider from "../../provider/authProvider";
import UserNotes from "../UserNotes";
//Smoke Test
test("renders UserNotes component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <UserNotes />
      </MemoryRouter>
    </AuthProvider>
  );
});

//Snapshot test
it("UserNotes matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <UserNotes />
      </MemoryRouter>
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});
