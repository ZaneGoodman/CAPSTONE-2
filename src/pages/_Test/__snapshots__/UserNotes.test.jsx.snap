// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotes matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Your Prayer Notes
  </h1>
  <ul />
</DocumentFragment>
`;
